src/main/java/net/dries007/tfc/common/blocks/devices/SheetPileBlock.java

```diff
@@ -0,0 +1,215 @@
+import net.dries007.tfc.common.blocks.ExtendedProperties;
+import net.dries007.tfc.util.Helpers;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A collection of arbitrary metal sheets in a single block
+ * Each side contains it's own sheet, which is identified by the face of the block which the side is on - so the UP property identifies the sheet on the TOP of the sheet pile block.
+ */
+public class SheetPileBlock extends ExtendedBlock implements EntityBlockExtension, DirectionPropertyBlock
+{
+    private static final Map<BooleanProperty, VoxelShape> SHAPES = new ImmutableMap.Builder<BooleanProperty, VoxelShape>()
+        .put(NORTH, box(0, 0, 0, 16, 16, 1))
+        .put(SOUTH, box(0, 0, 15, 16, 16, 16))
+        .put(EAST, box(15, 0, 0, 16, 16, 16))
+        .put(WEST, box(0, 0, 0, 1, 16, 16))
+        .put(UP, box(0, 15, 0, 16, 16, 16))
+        .put(DOWN, box(0, 0, 0, 16, 1, 16))
+        .build();
+
+    public static void removeSheet(Level level, BlockPos pos, BlockState state, Direction face, @Nullable Player player, boolean doDrops)
+    {
+        final BlockState newState = state.setValue(PROPERTY_BY_DIRECTION.get(face), false);
+
+        level.playSound(null, pos, SoundEvents.METAL_BREAK, SoundSource.BLOCKS, 0.7f, 0.9f + 0.2f * level.getRandom().nextFloat());
+        if (doDrops && (player == null || !player.isCreative()))
+        {
+            level.getBlockEntity(pos, TFCBlockEntities.SHEET_PILE.get()).ifPresent(pile -> {
+                final ItemStack stack = pile.removeSheet(face);
+                popResourceFromFace(level, pos, face, stack);
+            });
+        }
+
+        if (isEmpty(newState))
+        {
+            level.destroyBlock(pos, false);
+        }
+        else
+        {
+            level.setBlock(pos, newState, Block.UPDATE_CLIENTS);
+            level.levelEvent(LevelEvent.PARTICLES_DESTROY_BLOCK, pos, Block.getId(state));
+        }
+    }
+
+    public static void addSheet(LevelAccessor level, BlockPos pos, BlockState state, Direction face, ItemStack stack)
+    {
+        final BlockState newState = state.setValue(PROPERTY_BY_DIRECTION.get(face), true);
+
+        level.setBlock(pos, newState, Block.UPDATE_CLIENTS);
+        level.getBlockEntity(pos, TFCBlockEntities.SHEET_PILE.get()).ifPresent(pile -> pile.addSheet(face, stack));
+    }
+
+    /**
+     * @return The targeted face, if we can find one, or the first non-empty face, if we can find one, or {@code null}, if the block is empty.
+     */
+    @Nullable
+    public static Direction getTargetedFace(Level level, BlockState state, Player player)
+    {
+        final BlockHitResult result = Helpers.rayTracePlayer(level, player, ClipContext.Fluid.NONE);
+        if (result.getType() == HitResult.Type.BLOCK)
+        {
+            final Vec3 hit = result.getLocation();
+            @Nullable Direction firstDirection = null;
+            for (Map.Entry<BooleanProperty, VoxelShape> entry : SHAPES.entrySet())
```

Comment:
more of a comment than a review but I wonder if this is possible to do with comparing the difference between the hit vector and the block position to Direction's coordinate step field (ie, North and 0, 0, 1)

Reply:
possibly? but that feels like it would be more unreliable, the voxel shape method seems to work pretty good so far.